python/pyarrow/tests/test_parquet.py

```diff
@@ -157,6 +157,75 @@ def alltypes_sample(size=10000, seed=0, categorical=False):
     return pd.DataFrame(arrays)
 
 
+@pytest.mark.pandas
+@pytest.mark.parametrize('chunk_size', [1000])
+def test_get_record_batch_reader(tempdir, chunk_size):
+    df = alltypes_sample(size=10000, categorical=True)
+
+    filename = tempdir / 'pandas_roundtrip.parquet'
+    arrow_table = pa.Table.from_pandas(df)
+    assert arrow_table.schema.pandas_metadata is not None
+
+    _write_table(arrow_table, filename, version="2.0",
+                 coerce_timestamps='ms', chunk_size=chunk_size)
+
+    file_ = pq.ParquetFile(filename, batch_size=900)
+
+    def get_all_batches(f):
+        for row_group in range(f.num_row_groups):
+            batches = f.reader.get_batches(
+                [row_group],
+                range(len(f.schema))
+            )
+
+            dictionary_fields = {}
+
+            for batch in batches:
+
+                for idx, array in enumerate(batch):
+                    if isinstance(array.type, pa.DictionaryType):
+                        if array.dictionary:
+                            dictionary_fields[idx] = array.dictionary
+
+                if dictionary_fields:
+                    # if there's a dictionary field we need to fix
+                    # the categorical types here.
```

Comment:
It might be easier to test if you use a table/file without dictionaries (I don't think having a dictionary in there is essential to test the new batch functionality?)